distributed/tests/test_executor.py

```diff
@@ -3633,3 +3633,65 @@ def f():
 
                 future = e.submit(f)
                 assert future.result() == 1
+
+
+@gen_cluster(executor=False)
+def test_publish_simple(s, a, b):
+    e = Executor((s.ip, s.port), start=False)
+    yield e._start()
+    f = Executor((s.ip, s.port), start=False)
+    yield f._start()
+
+    data = yield e._scatter(range(3))
+    e.publish_dataset(data, 'data')
+    yield gen.sleep(0.2)
+    assert 'data' in s.published_data
```

Comment:
Tests that depend on sleep invariably fail some times.  Search for `assert time() < start` for an idiom to manage time-dependent tests.